File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.cpp
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        vector<int>L;
        
        for (auto &x : nums){
            if (L.empty() || L[L.size() - 1] < x){
                L.push_back(x);
            }
            else if(L[0] >= x){
                L[0] = x;
            }
            else{
                L[1] = x;
            }
            if(L.size() > 2) return true;
        }
        return L.size() > 2;
    }
};

File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        L = []
        
        for x in nums:
            if not L or L[-1] < x: L.append(x)
            elif L[0] >= x: L[0] = x
            else: L[1] = x
        return len(L) > 2